lib/logstash/outputs/opensearch/http_client/manticore_adapter.rb

```diff
@@ -27,13 +38,36 @@ def initialize(logger, options={})
       options[:cookies] = false
 
       @client_params = {:headers => DEFAULT_HEADERS.merge(options[:headers] || {})}
-      
+
+      if options[:auth_type] != nil && options[:auth_type]["type"] == "aws_iam"
+        aws_iam_auth_initialization(options)
+      end
+
       if options[:proxy]
         options[:proxy] = manticore_proxy_hash(options[:proxy])
       end
       
       @manticore = ::Manticore::Client.new(options)
     end
+
+    def aws_iam_auth_initialization(options)
+      @aws_default_port = 443
+      @aws_default_protocol = 'https'
+      @aws_default_region = 'us-east-1'
+      @aws_default_profile = 'default'
+
+      aws_access_key_id =  options[:auth_type]["aws_access_key_id"] || nil
+      aws_secret_access_key = options[:auth_type]["aws_secret_access_key"] || nil
+      session_token = options[:session_token] || nil
+      profile = options[:profile] || @aws_default_profile
+      instance_cred_retries = options[:instance_profile_credentials_retries] || 0
+      instance_cred_timeout = options[:instance_profile_credentials_timeout] || 1
+      @region = options[:auth_type]["region"] || @aws_default_region
+      @type = options[:auth_type]["type"]
```

Comment:
why do we need this variable?

Reply:
We will use it later in line 114

Reply:
But line 114 is outside this method. You can either define the variable outside and pass it to method or extract value separately.